library/spdm_requester_lib/libspdm_req_communication.c

```diff
@@ -13,6 +14,278 @@ libspdm_return_t libspdm_init_connection(void *spdm_context, bool get_version_on
 
     context = spdm_context;
 
+#if LIBSPDM_FIPS_MODE
+
+#if LIBSPDM_FIPS_TEST_AT_LOAD
+    if (context->crypto_context.tested_algo == 0) {
+        if (!libspdm_fips_run_selftest()) {
+            return LIBSPDM_STATUS_FIPS_FAIL;
+        } else {
+            /*all algos are tested and the all results passed*/
+            context->crypto_context.tested_algo = 0xFF;
```

Comment:
Why 0xFF?  There are more than 8 algorithms.  tested_algo is a uint32.

Reply:
Thanks. I have updated 0xFF -> 0xFFFF.

Reply:
Using 0xFFFF assumes the number of tests is 16.  i.e., if we add or remove selftests later, it needs to change here, and it is easy to miss.  
I'd suggest using OR result of all LIBSPDM_FIPS_SELF_TEST_XYZ instead of hardcoding to 0xFFFF.

Reply:
Thanks. I have changed the 0xFFFF by  using OR result of all LIBSPDM_FIPS_SELF_TEST_XYZ.